tests/serialized_test_data_generation/serialbox_to_netcdf.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import argparse
+import os
+import shutil
+import warnings
+import xarray as xr
+import f90nml
+import numpy as np
+sys.path.append("/serialbox/python")  # noqa: E402
+import serialbox  # noqa: E402
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        "converts serialbox data to netcdf"
+    )
+    parser.add_argument(
+        "data_path", type=str, help="path of serialbox data to convert",
+    )
+    parser.add_argument(
+        "output_path", type=str, help="output directory where netcdf data will be saved"
+    )
+    return parser
+
+
+def read_serialized_data(serializer, savepoint, variable):
+    data = serializer.read(variable, savepoint)
+    if len(data.flatten()) == 1:
+        return data[0]
+    return data
+
+
+def get_all_savepoint_names(data_path):
+    savepoint_names = set()
+    serializer = get_serializer(data_path, rank=0)
+    for savepoint in serializer.savepoint_list():
+        savepoint_names.add(savepoint.name)
+    return savepoint_names
+
+
+def get_serializer(data_path, rank):
+    return serialbox.Serializer(
+        serialbox.OpenModeKind.Read, data_path, "Generator_rank" + str(rank)
+    )
+
+
+def main(data_path: str, output_path: str):
+    os.makedirs(output_path, exist_ok=True)
+    namelist_filename_in = os.path.join(data_path, "input.nml")
+    namelist_filename_out = os.path.join(output_path, "input.nml")
+    if namelist_filename_out != namelist_filename_in:
+        shutil.copyfile(os.path.join(data_path, "input.nml"), namelist_filename_out)
+    namelist = f90nml.read(namelist_filename_out)
+    total_ranks = 6 * namelist["fv_core_nml"]["layout"][0] * namelist["fv_core_nml"]["layout"][1]
+    grid_list = []
+    for rank in range(total_ranks):
+        serializer = get_serializer(data_path, rank)
+        grid_savepoint = serializer.get_savepoint("Grid-Info")[0]
```

Comment:
Can you make writing `Grid-Info` optional in this script? For example, if we were to do this for the physics bucket, it will error here since there's no `Grid-Info` but it would be nice to still be able to convert the rest of the savepoints to netcdf files.

Reply:
I think I can delete the special handling for Grid-Info and treat it like any other savepoint, using the code below. Then it will only convert if the savepoint is present.

Reply:
Great, this also works!